<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>regexptest</class>
 <widget class="QMainWindow" name="regexptest">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>718</width>
    <height>539</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>regexptest</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="3" column="0">
     <widget class="QLabel" name="label">
      <property name="text">
       <string>\\</string>
      </property>
     </widget>
    </item>
    <item row="4" column="1">
     <widget class="QLineEdit" name="lineEdit_2"/>
    </item>
    <item row="4" column="0">
     <widget class="QLabel" name="label_2">
      <property name="text">
       <string>\</string>
      </property>
     </widget>
    </item>
    <item row="6" column="3">
     <widget class="QPlainTextEdit" name="plainTextEdit_2">
      <property name="maximumSize">
       <size>
        <width>200</width>
        <height>16777215</height>
       </size>
      </property>
     </widget>
    </item>
    <item row="3" column="3">
     <widget class="QPushButton" name="pushButton">
      <property name="maximumSize">
       <size>
        <width>200</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="text">
       <string>test</string>
      </property>
     </widget>
    </item>
    <item row="6" column="0" colspan="3">
     <widget class="QPlainTextEdit" name="plainTextEdit">
      <property name="plainText">
       <string>#include &lt;names.h&gt;				
#include “kpa_config.h”				
				
#define HW_SEQ_MKT_2    0x0002				
#define HW_SEQ_CTRL_ROOT    0x0004				
				
HW_SEQUENCE (HW_SEQ_CTRL_ROOT) {				
    KPA_2M2C_0:0, 				
    KPA_IP1_1:2,				
    KPA_IP1_1:3,				
    KPA_IP1_1:4,				
    @HW_SEQ_MKT_2      ; HW_SEQUENCE(HW_SEQ_MKT_2) becomes the part of the				
                                              ; current  sequence				
};				
				
HW_SEQUENCE (HW_SEQ_MKT_2) {				
    KPA_MKT_0:0,				
    KPA_MKT_0:1,				
    KPA_MKT_0:2,				
    KPA_MKT_0:3,				
    KPA_MKT_0:4,				
    KPA_MKT_0:5,				
    KPA_MKT_1:2,				
    KPA_MKT_1:3,				
    KPA_MKT_1:4				
};				
				
				
				
#define IP1_UPDATE EVENT_L00	

module  (  KPA_IP1  ,  SEQ_CTRL_ROOT    )       {




	procedure ( IP1_UPDATE) {			
		A := Ctrl;		
		Ctrl := A &amp; 0xFFFE;		
		rg[11] := Vx + dVx;		
		rg[10] := Vx - dVx;		
		rg[1] := 0xA710;		
		Ctrl := 0x0002 | iif ( A &amp; 0x0020, A | 0x8000, A);		
	}
}

module(KPA_IP2, HW_SEQ_CTRL){VAR_EVENTS{Ctrl ::: IP1_UPDATE}}

module	(	KPA_IP3	, 		 HW_CTRL_ROOT	)	 {	
	uword Vx : rg:7, dVx:rg:8;			
	uword Ctrl:port:0;			
	word A, B, C;			
}
				
module (KPA_IP4, HW_SEQ_CTRL_ROOT) {				
	uword Vx : rg:7, dVx:rg:8;			
	uword Ctrl:port:0;			
	word A, B, C;			
				
	VAR_EVENTS {			
		Ctrl ::: IP1_UPDATE		
	}			
				
	procedure ( IP1_UPDATE) {			
		A := Ctrl;		
		Ctrl := A &amp; 0xFFFE;		
		rg[11] := Vx + dVx;		
		rg[10] := Vx - dVx;		
		rg[1] := 0xA710;		
		Ctrl := 0x0002 | iif ( A &amp; 0x0020, A | 0x8000, A);		
	}			
				
	procedure Run (EVENT_GRUN) {			
		Ctrl := 0;		
		Vx := 0;		
		dVx := 0;		
		B := 0;		
		while (1) {		
			rg[3+B] := 0;	
			if (B &gt; 5)	
				break;
			else	
				B := B + 1;
			if (Ctrl &amp; 0x0200) {	
				local_send_event (IP1_UPDATE);
				end_proc ();
			}	
		}		
		debug_prnt (&quot;Module IP1 was run at %02d:%02d&quot;, m_pos, m_channel);		
	}			
}				
</string>
      </property>
     </widget>
    </item>
    <item row="2" column="1" colspan="3">
     <widget class="QPlainTextEdit" name="plainTextEdit_3">
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>200</height>
       </size>
      </property>
     </widget>
    </item>
    <item row="3" column="1" colspan="2">
     <widget class="QLineEdit" name="lineEdit">
      <property name="toolTip">
       <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Ok&lt;/p&gt;&lt;p&gt;Ok2&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
      <property name="text">
       <string>module[\s\t]*\([\s\t]*([\w]*)[\s\t]*\,[\s\t]*([\w]*)[\s\t]*\) </string>
      </property>
     </widget>
    </item>
    <item row="0" column="1" colspan="2">
     <widget class="QLabel" name="label_3">
      <property name="text">
       <string>History:                                                                QREGEXP TEST</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>718</width>
     <height>19</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
